src/app/test-record/test-record.container.spec.ts

```diff
@@ -43,18 +52,46 @@ describe('TestRecordContainer', () => {
   }));
 
   beforeEach(() => {
+    store = TestBed.get(Store);
+    spyOn(store, 'dispatch').and.callThrough();
     fixture = TestBed.createComponent(TestRecordContainer);
     container = fixture.componentInstance;
-    container.testRecordObservable$ = of(testRecord);
-    container.testTypeNumber = 'W01A34247';
-    container.seatBeltApplicable = ['11'];
-    container.emissionDetailsApplicable = ['22'];
-    container.defectsApplicable = ['33'];
+    container.testRecord$ = of(testRecordTestType);
+    container.testTypesApplicable = {} as TestTypesApplicable;
     fixture.detectChanges();
   });
 
   it('should create', () => {
     expect(container).toBeTruthy();
     expect(fixture).toMatchSnapshot();
   });
+
+  it('should set form state', () => {
+    container.setFormState(true);
+    expect(container.isFormDirty).toBeTruthy();
+  });
+
+  it('should dispatch action to set form state', () => {
+    container.currentStateHandler(VIEW_STATE.VIEW_ONLY);
+    expect(store.dispatch).toHaveBeenCalled();
+    expect(store.dispatch).toHaveBeenCalledWith({
+      editState: 0,
+      type: '[TestResultComponent] SetCurrentState'
```

Comment:
Same let's use the enum we have please so if we change it later, the test shouldn't break for no valid reason.
Same for VIEW_STATE.VIEW_ONLY